app/Http/Middleware/LogAction.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+
+class LogAction
+{
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (Auth::check() && Auth::user()->isAdmin() && $request->method() !== 'GET') {
+            parse_str($request->getContent(), $body);
+            $bodyKeys = [];
+            foreach ($body as $key => $value) {
+                array_push($bodyKeys, $key); // Only include keys from request body to avoid logging any sensitive values.
+            }
+            $logArray = [
+                'message' => 'Admin non-GET request',
+                'admin_user_id' => $request->user()->id,
+                'url' => $request->url(),
+                'verb' => $request->method(),
+                'body' => $bodyKeys
```

Comment:
Maybe change this key from "body" to "bodyKeys", to be more clear about whats actually being logged.

Reply:
@tristan-orourke I've changed it to `body_keys` to match the snake case of `admin_user_id`. If you'd prefer camel case, let me know and I can the keys accordingly, thanks.